Assets/Scripts/Escape/EscapeDetectorBase.cs

```diff
@@ -0,0 +1,14 @@
+using UnityEngine;
+
+// Base implementation for an escape detector.
+public abstract class EscapeDetectorBase : IEscapeDetector {
+  // Agent that is pursuing the target.
+  public IAgent Agent { get; init; }
+
+  public EscapeDetectorBase(IAgent agent) {
+    Agent = agent;
+  }
```

Comment:
All three escape detectors check for null target but dont check if `Agent` is null or the `Agent.StaticConfig` is null.
Low hanging fruit null check might be here with
```csharp
public EscapeDetectorBase(IAgent agent) {
  if (agent == null) {
    throw new ArgumentNullException(nameof(agent));
  }
  Agent = agent;
}
```

Reply:
I'd even say that a precondition is that `agent` is non-null.  Otherwise, throwing the exception here is equivalent to a NRE being thrown in the code.  All the hierarchical algorithms actually assume a non-null Agent - if you have any ideas on how to take care of all of them checking for non-null, that might be a cleaner idea (than a null check everywhere).